### PR TITLE
86-upload-file-by-url-cli

### PR DESCRIPTION
**Pull request by klavinsjanis**

### CODE REVIEW

**Review by DarjaLeonova — APPROVED** [review]

File: cmd/client/cmd/upload.go [hunk 1]
```diff
@@ -46,9 +51,16 @@ func newUploadCmd() *cobra.Command {
 				return fmt.Errorf("upload file: get cli parameter 'file': %w", err)
 			}
 
-			data, err := os.ReadFile(filePath)
-			if err != nil {
-				return fmt.Errorf("upload file: read file from path: %w", err)
+			var data []byte
+
+			if strings.HasPrefix(filePath, "http") || strings.HasPrefix(filePath, "https") {
```
Comment (janishorsts) [inline cmd/client/cmd/upload.go:56]: include `://` - `http://` and `https://`. The folder name can start with `http`.

File: cmd/client/cmd/upload.go [hunk 2]
```diff
@@ -100,3 +112,35 @@ func newUploadCmd() *cobra.Command {
 
 	return uploadCmd
 }
+
+// readFileFromURL reads translation file from URL.
+func readFileFromURL(ctx context.Context, filePath string) ([]byte, error) {
+	u, err := url.ParseRequestURI(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("parse URL: %w", err)
```
Comment (janishorsts) [inline cmd/client/cmd/upload.go:120]: Your intent is not to parse, but to validate

```suggestion
		return nil, fmt.Errorf("validate file URL: %w", err)
```

File: cmd/client/cmd/upload.go [hunk 3]
```diff
@@ -100,3 +112,35 @@ func newUploadCmd() *cobra.Command {
...
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("prepare new GET request: %w", err)
+	}
+
+	resp, err := otelhttp.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("do GET request: %w", err)
```
Comment (janishorsts) [inline cmd/client/cmd/upload.go:130]:
```suggestion
		return nil, fmt.Errorf("fetch file: %w", err)
```

File: cmd/client/cmd/upload.go [hunk 4]
```diff
@@ -100,3 +112,35 @@ func newUploadCmd() *cobra.Command {
...
+	}
+
+	defer resp.Body.Close()
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("bad status: %s", resp.Status)
```
Comment (janishorsts) [inline cmd/client/cmd/upload.go:137]:
```suggestion
		return nil, fmt.Errorf("fetch file status: got %s, expected 200", resp.Status)
```

File: cmd/client/cmd/upload.go [hunk 5]
```diff
@@ -73,7 +85,7 @@ func newUploadCmd() *cobra.Command {
 
 	uploadFlags := uploadCmd.Flags()
 	uploadFlags.String("serviceID", "", "service UUID")
-	uploadFlags.String("file", "", "file path")
+	uploadFlags.String("file", "", "translation file local path or URL")
```
Comment (janishorsts) [inline cmd/client/cmd/upload.go:88]:
```suggestion
	uploadFlags.String("file", "", "local path or URL for the translation file")
```

**Review by janishorsts — APPROVED** [review]

**Pull request merged**